TheJumpCloud/support
# JumpCloud Module 1.21.0

## Issues
* [SA-2540](https://jumpcloud.atlassian.net/browse/SA-2540) - Add revoveryEmail parameter to New/Set/Get-JCUser
* [SA-2539](https://jumpcloud.atlassian.net/browse/SA-2539) - Update-JCModule fix for installing new production SDKs


## What does this solve?

New/Set/Get-JCUser should have the option to set the `recoveryEmail` parameter. This release addresses this missing parameter, thanks to #372 by @rybai for the suggestion.

Upon updating new SDK versions from PowerShell Gallery, Update-JCModule would throw an error stating that it could never update the existing SDKs. This is in short an issue with namespaces. If one SDK is loaded into memory and updated, PowerShell will throw an error stating that the module could not be updated with the `Update-Module` function. Instead we opted to use `Install-Module` to install two separate instances of the SDK module, then uninstall versions which were outdated.

## Is there anything particularly tricky?

N/A

## How should this be tested?

1.  Import the branch module: Import-Module ".../support/PowerShell/JumpCloud Module/JumpCloud.psd1"
2. Create new user with recovery email param: `New-JCUser -firstname test -lastname recovery -username testRecoveryuser -email test@testdelete.com -recoveryEmail testRecoveryEmail@testdeleteme.com`
3.  `Get-JCUser -recoveryEmail testRecoveryEmail@testdeleteme.com`
4. `Set-JCUser -username testRecoveryuser -recoveryEmail newrecoveryemail@testdeleteme.com`
5. Lastly, verify: `Get-JCUser -recoveryEmail newrecoveryemail@testdeleteme.com`

See #373 for testing SDK update workflow

## Screenshots
